### PR TITLE
Only load what is read.

### PR DESCRIPTION
**Pull request by PapyChacal**
The codegen used to generates `stencil.load`s for any field that wasn't written to in a given equations, leading to superfluous `load`/`store` clashes when dealing with multiple equations. Now, only `load` the actual inputs to the equation at hand.

### CODE REVIEW

File: tests/test_xdsl_op_correctness.py [hunk 1]
```diff
@@ -64,6 +64,4 @@ def test_u_and_v_conversion():
     norm_v2 = norm(v)
 
     assert np.isclose(norm_u, norm_u2, atol=1e-5, rtol=0)
-    assert np.isclose(norm_u, 26.565891, atol=1e-5, rtol=0)
```
Comment (georgebisbas) [inline tests/test_xdsl_op_correctness.py:67]: What is the norm now? Can you add this back with the right number?

File: tests/test_xdsl_op_correctness.py [hunk 2]
```diff
@@ -41,16 +41,16 @@ def test_u_plus1_conversion():
...
     assert np.isclose(norm1, 23.43075, atol=1e-5, rtol=0)
 
-@pytest.mark.xfail(reason="Needs a fix in offsets")
+
 def test_u_and_v_conversion():
     # Define a simple Devito Operator
     grid = Grid(shape=(3, 3))
-    u = TimeFunction(name='u', grid=grid, time_order=2)
-    v = TimeFunction(name='v', grid=grid, time_order=2)
+    u = TimeFunction(name='u', grid=grid, time_order=3)
```
Comment (georgebisbas) [inline tests/test_xdsl_op_correctness.py:48]: why time_order =3?

File: tests/test_xdsl_op_correctness.py (hunk 1 above)
Comment (PapyChacal) [reply tests/test_xdsl_op_correctness.py:67]: Just felt strange to me, added it back, it's now 2.0664787

File: tests/test_xdsl_op_correctness.py (hunk 2 above)
Comment (PapyChacal) [reply tests/test_xdsl_op_correctness.py]: Just a leftover from exploring; I tried examples that were doing in-place when using time_order 2.
Switched this example back to time_order 2.

File: tests/test_xdsl_op_correctness.py (hunk 1 above)
Comment (PapyChacal) [reply tests/test_xdsl_op_correctness.py:67]: 2.0664592 after switching back to time_order 2 :wink:

**Review by georgebisbas — COMMENTED** [review]
I think we are prefect here, just need one effort on my side to play a bit locally and then will formally approve!

**Review by georgebisbas — APPROVED** [review]

**Pull request merged**